applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -253,6 +327,350 @@ void NavierStokesWallCondition<3,3>::CalculateNormal(array_1d<double,3>& An )
 }
 
 
+
+/**
+ * @see NavierStokesWallCondition::CalculateBehrSlipLeftHandSideContribution
+ * 
+ * @tparam TDim 
+ * @tparam TNumNodes 
+ */
+
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim,TNumNodes>::CalculateBehrSlipLeftHandSideContribution(  MatrixType& rLeftHandSideMatrix,
+                                                                                            ProcessInfo& rCurrentProcessInfo,
+                                                                                            ConditionDataStruct& rDataStruct )
+{
+    KRATOS_TRY
+    // check if already computed and returning the stored value
+    // possible for static meshes - not general
+    if ( mHasBehrSlipLeftHandSideMatrix == true ){
```

Comment:
Since the leftHandSide does only depend on the mesh geometry, my thought was to avoid a recomputation and to store it, instead. Considering memory consumption, it may be better to recompute it. What would be your preference here?

Reply:
In general, recompute. Is that expensive to calculate, compared to a regular FEM matrix?

Reply:
After having another look, I do not think that it is considerably more expensive. Accordingly, - if nobody has any doubts - I will remove this check and the stored matrix such that I re-computation is conducted every time.

Reply:
The recomputation was implemented. Thank you.